### DISCUSSION TITLE
How to get the JWT in a NextJS Route Handler

### DISCUSSION THREAD

**Discussion by Charismara** [nextauthjs/next-auth#12108] [Help]
Hi there,

I am currently upgrading from Next Auth V4 to V5 and struggle on getting the JWT Token in my Route Handler.

In V4 I've added Access Tokens, Refresh Tokens and other credentials though the JWT Callback to Next Auths JWT Token. Those credentials then got attached to the client browser as http only cookies though a custom route handler. 

That system worked well for us and i would love to use it in V5 but haven't found a way to load the JWT information instead of the session in route handlers.

Has someone figured out how to access the JWT?

**Comment by k-taro56** [comment]
Sorry, it is no longer available.

**Reply by idc77** [reply]
why are you trolling? why is this discussion forum full of trolls?

**Reply by Charismara** [reply]
> why are you trolling? why is this discussion forum full of trolls?

Is there a way in V5 to access the JWT in a NextJS route handler? 👀
